Overhaul `README` and documentation catalog to introduce best practices (#363)
apple/swift-log: Sources/Logging/Docs.docc/index.md

# ``Logging``

A unified, performant, and ergonomic logging API for Swift.

## Overview

SwiftLog provides a logging API package designed to establish a common API the
ecosystem can use. It allows packages to emit log messages without tying them to
any specific logging implementation, while applications can choose any
compatible logging backend.

SwiftLog is an _API package_ which cuts the logging problem in half:
1. A logging API (this package)
2. Logging backend implementations (community-provided)

This separation allows libraries to adopt the API while applications choose any
compatible logging backend implementation without requiring changes from
libraries.

## Getting Started

> If you have a Swift application, or a library, targeting this logging API is
> recommended.

Reviewer: ```suggestion
Use this If you are writing a cross-platform application (for example, Linux and macOS) or library, target this logging API.
```

Makes this far more direct. Too direct?

### Adding the Dependency

Add the dependency to your `Package.swift`:

```swift
.package(

>     url: "https://github.com/apple/swift-log.git",

Reviewer 2: ```suggestion
    url: "https://github.com/apple/swift-log",
```

Shorter better

    from: "1.6.0"
)
```

And to your target:

```swift
.target(
    name: "YourTarget",
    dependencies: [
        .product(name: "Logging", package: "swift-log")
    ]
)
```

### Basic Usage

```swift
// Import the logging API
import Logging

// Create a logger with a label
let logger = Logger(label: "MyLogger")

// Use it to log messages
logger.info("Hello World!")
```

> This will output:

Reviewer: ```suggestion
This outputs:
```

using present tense, not future

```
2019-03-13T15:46:38+0000 info: Hello World!
```

### Default Behavior

SwiftLog provides basic console logging via ``StreamLogHandler``. By default it
uses `stdout`, however, you can configure it to use `stderr` instead:

```swift
LoggingSystem.bootstrap(StreamLogHandler.standardError)
```

> `StreamLogHandler` is primarily for convenience. For production applications,

Reviewer 2: ```suggestion
``StreamLogHandler`` is primarily for convenience. For production applications,
```

> implement the `LogHandler` protocol directly or use a community-maintained

Reviewer 2: ```suggestion
implement the ``LogHandler`` protocol directly or use a community-maintained
```
